### PR TITLE
Binoculars, Map mods: On use, trigger 'on_rightclick' of pointed node

### LINKED ISSUE #2241: binoculars doesn't delegate on_use to nodes
just like the bucket and a few more items, the binoculars should be allowed placement in itemframes etc. The current code doesn't check that the pointed node has a handler and always makes it toggle zoom modes.

https://github.com/minetest/minetest_game/blob/0ea6065a09f1fee014c7f2a9a36605d57986a209/mods/binoculars/init.lua#L61

### PR DESCRIPTION
**Pull request by paramat**
WIP in testing, code needs update.
Closes #2241 
@sofar

### CODE REVIEW

**Comment by sofar** [comment]
yeah, that looks right... almost seems worth to make a function for this...

File: mods/binoculars/init.lua [hunk 1]
```diff
@@ -59,6 +59,15 @@ minetest.register_craftitem("binoculars:binoculars", {
 	stack_max = 1,
 
 	on_use = function(itemstack, user, pointed_thing)
+		if pointed_thing then
```
Comment (SmallJoker) [inline mods/binoculars/init.lua]: `pointed_thing` is always specified. Instead, check whether the player used the binoculars on a node: `pointed_thing.type == "node"`.

Comment (paramat) [reply mods/binoculars/init.lua]: Ok.

Comment (paramat) [reply mods/binoculars/init.lua]: However we're waiting to see if this PR is actually needed, see issues.

**Comment by paramat** [comment]
PR not needed, see https://github.com/minetest/minetest_game/issues/2241#issuecomment-432585690

**Pull request closed without merging**